src/main/java/net/fabricmc/loom/util/fmj/FabricModJsonFactory.java

```diff
@@ -167,7 +167,10 @@ public static boolean isModJar(File file, ModPlatform platform) {
 
 	public static boolean isModJar(Path input, ModPlatform platform) {
 		if (platform == ModPlatform.FORGE) {
-			return ZipUtils.contains(input, "META-INF/mods.toml");
+//			return ZipUtils.contains(input, "META-INF/mods.toml");
+			// Forge don't care if a nested jar contains mod manifest.
+			// Manifest will make the language provider jar loaded to GAME layer
+			return true;
```

Comment:
This is wrong, this method is also used elsewhere (eg `AccessWidenerUtils`) to check for mods. You should patch the relevant call sites instead.